### PR TITLE
RELEASES.md: Add 1.58 release note for `File::options` stabilization

### PR DESCRIPTION
**Pull request by joshtriplett**

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -42,6 +42,7 @@ Stabilized APIs
 - [`{integer}::saturating_div`]
 - [`Option::unwrap_unchecked`]
 - [`NonZero{unsigned}::is_power_of_two`]
+- [`File::options`] - Equivalent to `OpenOptions::new`, creating an instance of the `OpenOptions` builder. Instead of `OpenOptions::new().append(true).open("example.log")` you can write `File::options().append(true).open("example.log")`. This also avoids the need to import `OpenOptions`.
```
Comment (pietroalbini) [inline RELEASES.md:45]: Hmm, I don't recall having long descriptions as part of the stabilized apis section. @Mark-Simulacrum?

Comment (joshtriplett) [reply RELEASES.md:45]: Happy to put it elsewhere, but I think an explanation seems useful here.

Comment (joshtriplett) [reply RELEASES.md:45]: In particular, we commonly put explanations in the "Libraries" section, and this seems similar.

Comment (camelid) [reply RELEASES.md:45]: The docs for the function already have this info AFAICT; maybe that's good enough?

Comment (Mark-Simulacrum) [reply RELEASES.md:45]: I'm not a huge fan when it's not relatively important information (e.g., compatibility notes or similar) -- particularly as this is replicating linked documentation it doesn't feel necessary to me, and seems weird to do solely for this API, so I'd prefer to drop it for consistency.

Comment (joshtriplett) [reply RELEASES.md:45]: Alright, I'll work on improving the existing linked documentation instead.

**Comment by joshtriplett** [comment]
@pietroalbini Done.

**Pull request merged**